src/tls13.c

```diff
@@ -9293,6 +9573,157 @@ static int DoTls13CertificateVerify(WOLFSSL* ssl, byte* input,
                 ERROR_OUT(BUFFER_ERROR, exit_dcv);
             }
 
+#ifdef WOLFSSL_DUAL_ALG_CERTS
+            if (!wolfSSL_is_server(ssl) &&
+                (ssl->sigSpec != NULL) &&
+                (*ssl->sigSpec != WOLFSSL_CKS_SIGSPEC_NATIVE)) {
+                word16 sa;
+                int keyRet;
+                word32 tmpIdx = 0;
+
+                /* ssl->peerCert->sapkiDer is the alternative public key.
+                 * Convert it into a usable public key. */
+                if (args->altSigAlgo == 0)
+                    sa = args->sigAlgo;
+                else
+                    sa = args->altSigAlgo;
+
+                switch (sa) {
```

Comment:
Messy having switch after switch.

Reply:
Yes, but if we combined all the switches it would result in a lot of duplicated code.  I'm going to leave it as is.

Reply:
Extract the Dilithium work into a separate function and the Falcon work in to another.
Then pass the level to the new functions.
Only extract the first 3 switch statements.

Reply:
@SparkiDev : Wow....what a difference it makes!! Thanks for the catch.  Fixed!